apollo-federation/src/query_plan/fetch_dependency_graph.rs

```diff
@@ -2676,10 +2677,9 @@ impl FetchDependencyGraphNode {
         let operation =
             operation_compression.compress(&self.subgraph_name, subgraph_schema, operation)?;
         let operation_document = operation.try_into().map_err(|err| match err {
-            FederationError::SingleFederationError {
-                inner: SingleFederationError::InvalidGraphQL { diagnostics },
-                ..
-            } => FederationError::internal(format!(
+            FederationError::SingleFederationError(SingleFederationError::InvalidGraphQL {
+                diagnostics,
+            }) => FederationError::internal(format!(
```

Comment:
```suggestion
            }) => internal_error!(
```

Reply:
I was not able to commit your suggestions via github, not sure why, so it all had to be in a manual commit 7d629059cad60c4f8ccf514607e08fa059a7a4da